### PR TITLE
adapt frontend to new db schema

### PR DESCRIPTION
**Pull request by tatchi**
I've noticed that the frontend was currently broken due to the db schema changes introduced in #22 

This PR makes the changes to adapt the frontend to this new schema. There are no errors anymore but I'm not yet very familiar with the application so I might have missed some stuff.

### CODE REVIEW

File: frontend-new/src/App.res [hunk 1]
```diff
@@ -93,11 +85,14 @@ let make = () => {
 
   // Fetch benchmarks data
   let ({ReasonUrql.Hooks.response: response}, _) = {
-    let startDate = (Js.Date.getTime(startDate) /. 1000.0)->Js.Json.number
-    let endDate = (Js.Date.getTime(endDate) /. 1000.0)->Js.Json.number
+    let startDate = Js.Date.toISOString(startDate)->Js.Json.string
+    let endDate = Js.Date.toISOString(endDate)->Js.Json.string
```
Comment (tatchi) [inline frontend-new/src/App.res:89]: I'm definitely not good at handling dates 😅 Is that the correct format?

File: frontend-new/src/Components.res [hunk 2]
```diff
@@ -573,7 +573,7 @@ module SelectX = {
     <Select ?name ?sx ?disabled ?id ?placeholder ?value onChange>
       {options
       |> Array.mapi((i, opt) => <option key={string_of_int(i)}> {Rx.string(encode(opt))} </option>)
-      |> Rx.array}
+      |> (el => Rx.array(el))}
```
Comment (tatchi) [inline frontend-new/src/Components.res:576]: To fix a warning at compilation

File: frontend-new/src/Components.res [hunk 3]
```diff
@@ -680,7 +680,7 @@ module Switch = {
   @react.component
   let make = (~name=?, ~onToggle=?, ~on as checked=?) => {
     let onChange = switch onToggle {
-    | Some(f) => Some(event => f())
+    | Some(f) => Some(_ => f())
```
Comment (tatchi) [inline frontend-new/src/Components.res:683]: Also to fix a warning

File: frontend-new/src/App.res [hunk 4]
```diff
@@ -26,24 +25,17 @@ let collectBranches = (data: array<GetBenchmarks.t_benchmarks>) => {
...
-      metrics: result
-      ->jsonFieldExn("metrics", Js.Json.Object)
-      ->jsDictToMap
-      ->Belt.Map.String.map(v => BenchmarkTest.decodeMetricValue(v)),
-      commit: item.commits,
-    }
-  })
+let getTestMetrics = (item: GetBenchmarks.t_benchmarks): BenchmarkTest.testMetrics => {
+  {
+    BenchmarkTest.name: item.repo_id,
```
Comment (rizo) [inline frontend-new/src/App.res:30]: Just a note: `BenchmarkTest.name` should ideally be mapped to the `benchmarks.test_name` SQL column. I haven't applied the migration that adds this column yet, so let's keep it like this in this PR.

File: frontend-new/src/App.res (hunk 1 above)
Comment (rizo) [reply frontend-new/src/App.res:89]: In this case we're to follow the instructions for Hasura-Postgresql dates: https://hasura.io/blog/postgres-date-time-data-types-on-graphql-fd926e86ee87/

But, yes. ISO8601 is what we need here.

**Review by rizo — APPROVED** [review]

**Comment by rizo** [comment]
Could you please merge this @gs0510?

**Comment by gs0510** [comment]
Can you rebase @rizo? I'll merge once you do. Thanks!

**Comment by tatchi** [comment]
@rizo @gs0510 I merged tatchi/current-bench#1 and rebased. Should be good to merge :)

**Pull request merged**